fuel/datasets/hdf5.py

```diff
@@ -411,21 +616,27 @@ def _in_memory_get_data(self, state=None, request=None):
         return data, shapes
 
     def _out_of_memory_get_data(self, state=None, request=None):
+        if not isinstance(request, (slice, list)):
```

Comment:
That would be also cool for making the code simpler, because as now it have quite a lot of nested ifs.